src/react/DevicesPage.jsx

```diff
@@ -35,126 +36,150 @@ import SingleCardPage from './ui/SingleCardPage';
 import { useAlerts } from './AlertManager';
 
 const DEVICES_PER_PAGE = 20;
+const DEVICES_PER_REQUEST = 100;
 const MAX_SHOWN_PAGES = 10;
 
+const matchFilters = (device, filters) => {
```

Comment:
This is already fine. However I know there's a nice pattern to put the "data" last in function's arguments, so that the function can be easily curried / partially applied. What do you think?

Reply:
Depending on the language you use you'll see different patterns, some use data first and others data last.
If we do chose one over the other, we should stick with it for consistency, but I don't have strong opinions on which one to chose.
In JS there is no native currying or partial applying. You have to wrap your own function every time.
PS: One could say JS has a "native" way to curry parameters with the method `bind` but it affect the `this` parameter unless you always pass a `null` argument, leaving the code harder to read.